### ISSUE TITLE
Fix for "You do not have permission to download this file" due to Cloudflare or firewall plugins query string modification

### ISSUE THREAD

**Issue by sergeykomlev** [awesomemotive/easy-digital-downloads#9345]
I found that this issue occurs in my system because of the following line and not because of query string order like in #8917.

https://github.com/awesomemotive/easy-digital-downloads/blob/0c0bcd9579ca8b150196509c22d07b6c13bb84a3/includes/process-download.php#L854

Looks like Cloudflare or firewall plugins remove `:` symbol from `eddfile` string. When the function `parse_url` is used with an empty array as the function argument purified string `eddfile` is used. 

So instead of `111111:22222:3:4` it's converted to `1111112222234` and this breaks all of my download links because tokens going to be different.

Solution is add `$args` to function `parse_url`.

```php
$parts = parse_url( add_query_arg( $args ) );
```

**Comment by shazahm1** [comment]
I started experiencing this issue Saturday morning, at least that is when I received my first customer report. I submitted a support ticket to EDD, so hopefully, they will take a look immediately. My quick fix since I had stuff to do on Saturday was to override `edd_validate_url_token()` and set `$valid_token` to true. I'll try your patch since it restores the download links' security.

**Comment by shazahm1** [comment]
@sergeykomlev 

Do you happen to have Anti-Spam by CleanTalk version 5.182.1, released on Aug 04, 2022, installed? I did. According to their changelog, it seems they added some server variable sanitization. Not sure exactly what they mean by that, but if I deactivate that plugin, the issue goes away. It seems like they might be manipulating the GET request variables.

Updating to version 5.182.2, released on Aug 08, 2022, corrects this issue for me.

Perhaps EDD should use `filter_input()` instead of accessing the GET superglobal directly, which should/would have prevented this.

**Comment by sergeykomlev** [comment]
> Updating to version 5.182.2, released on Aug 08, 2022, corrects this issue for me.

Yes, I also have Anti-Spam by CleanTalk installed. I will update this plugin to the latest version. 

Thanks for your report, @shazahm1!

**Comment by sergeykomlev** [comment]
@shazahm1, I deactivated plugin Security & Malware scan by CleanTalk and my website started to working faster and better. I found that Cloudflare have a better protection system and managed WAF  rules helps me prevent DDOS attacks on my servers on DNS level.

**Comment by cklosowski** [comment]
As this is a file delivery ticket we'll have to take extra precaution around any changes. The switch to `filter_input` seems like a good start, however we'll have to look into any possible adjustments to the `$_GET` variables in extensions and backwards compatibility checks.

Per the PHP doc notes:

> Note that this function doesn't (or at least doesn't seem to) actually filter based on the current values of $_GET etc. Instead, it seems to filter based off the original values.

Which means that if we manually alter the super global for `$_GET` in anyway, `filter_input` will not catch it.
